src/registered.rs

```diff
@@ -52,6 +54,65 @@ impl Context for Registered {
     const CKD_DOMAIN: HardenedOnlyCkdDomain = PrfExpand::REGISTERED_ZIP32_CHILD;
 }
 
+/// An error that occurred in cryptovalue derivation.
+#[derive(Debug, Clone, Copy)]
+pub enum DerivationError {
+    /// The provided seed data was invalid. A seed must be between 32 and 252 bytes in length,
+    /// inclusive.
+    SeedInvalid,
+    /// The provided context string is invalid; context strings must be no greater than 252 bytes
+    /// in length.
+    ContextStringInvalid,
+    /// The provided subpath was empty; empty subpaths are not permitted by this API as deriving
+    /// at the empty subpath would expose the master key.
```

Comment:
```suggestion
    /// The provided subpath was empty. Empty subpaths are not permitted by this API, as the
    /// full-width cryptovalue at the empty subpath would be outside the allowed subtree
    /// rooted at $m_{context} / ZipNumber'$.
```
This error only occurs for `cryptovalue_from_subpath`, and it would not leak the master key if it were allowed.